### PR TITLE
hack/tools:  update to  golangci-lint v1.56.2 and Go v1.21

### PR DESCRIPTION
**Pull request by yankay**
#### What this PR does / why we need it:

The golangci-lint v1.56.0 supports Go 1.22, related to:  https://github.com/kubernetes/release/issues/3280 .
- The differences: https://github.com/golangci/golangci-lint/compare/v1.55.2...v1.56.2
- The release notes:  https://github.com/golangci/golangci-lint/releases/tag/v1.56.2

Because the golangci-lint v1.56.0  requires go-exhaustruct/v3@v3.2.0, and it requires go@1.21, so update tools  golang to 1.21

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?
```
NONE
```

### CODE REVIEW

**Comment by hwdef** [comment]
why not go 1.22?

File: hack/tools/go.mod [hunk 1]
```diff
@@ -1,13 +1,13 @@
 module k8s.io/kubernetes/hack/tools
 
-go 1.19
+go 1.21
```
Comment (BenTheElder) [inline hack/tools/go.mod:3]: hmm @liggitt probably we should be forcing this to stay in sync with the root go.mod?

**Comment by BenTheElder** [comment]
> why not go 1.22?

We're not using go1.22 yet for the language version: https://github.com/kubernetes/kubernetes/blob/5d39bc5680c3c2b2c6e9bfcad5e2adededf4db28/go.mod#L9

As opposed to the *toolchain* version: 
https://github.com/kubernetes/kubernetes/blob/5d39bc5680c3c2b2c6e9bfcad5e2adededf4db28/.go-version#L1

**Comment by BenTheElder** [comment]
For future reference: the release note label is not working because the code block must be labeled release-note as it was originally like
> \```release-note

In this case, we can instead do
/release-note-none

**Review by BenTheElder — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**